src/components/fields/ObjectField.js

```diff
@@ -84,67 +81,124 @@ class ObjectField extends Component {
-              onChange={this.onPropertyChange(name)}
-              onBlur={onBlur}
-              registry={this.props.registry}
-              disabled={disabled}
-              readonly={readonly}/>
-          );
-        })
-      }</fieldset>
-    );
+          <fieldset>
+              {title ? <TitleField
+                      id={`${idSchema.$id}__title`}
+                      title={title}
+                      required={required}
+                      formContext={formContext}/> : null}
+              {schema.description ?
+                  <DescriptionField
+                      id={`${idSchema.$id}__description`}
+                      description={schema.description}
+                      formContext={formContext}/> : null}
+              {orderedProperties.map((name, index) => {
+                return (
+                      <SchemaField key={index}
+                                   name={name}
+                                   required={this.isRequired(name)}
+                                   schema={schema.properties[name]}
+                                   uiSchema={uiSchema[name]}
+                                   errorSchema={errorSchema[name]}
+                                   idSchema={idSchema[name]}
+                                   formData={this.state[name]}
+                                   onChange={this.onPropertyChange(name)}
+                                   onBlur={onBlur}
+                                   registry={this.props.registry}
+                                   disabled={disabled}
+                                   readonly={readonly}/>
+                  );
+              })}
+          </fieldset>
+      );
+  }
+
+  renderAdditionalProperties(schema, title){
+    const {
+          uiSchema,
+          errorSchema,
+          idSchema,
+          required,
+          disabled,
+          readonly,
+          onBlur
+      } = this.props;
+    const {fields, formContext} = this.props.registry;
+    const {SchemaField, TitleField, DescriptionField} = fields;
+
+    return (
+          <fieldset>
+              {title ? <TitleField
+                      id={`${idSchema.$id}__title`}
+                      title={title}
+                      required={required}
+                      formContext={formContext}/> : null}
+              {schema.description ?
+                  <DescriptionField
+                      id={`${idSchema.$id}__description`}
+                      description={schema.description}
+                      formContext={formContext}/> : null}
+              {Object.keys(this.state).map((name, index) => {
+                const childIdSchema = {"$id": `${idSchema.$id}__${name}`};
+                return (
+                      <SchemaField key={index}
+                                   name={name}
+                                   required={this.isRequired(name)}
+                                   schema={schema.additionalProperties}
```

Comment:
I'm pretty sure this doesn't correctly support a schema that has both `properties` and `additionalProperties`.